nodejs/week2/app.js

```diff
@@ -0,0 +1,99 @@
+const express = require('express')
+const app = express()
+const port = process.env.PORT || 3000
+const documents = require('./documents.json')
+
+// Support parsing JSON requests
+app.use(express.json())
+
+app.get('/', (req, res) => {
+    res.send('This is a search engine')
+})
+
+//GET /search
+app.get('/search', (req,res) => {
+    try {
+        const queryParam = req.query.q;
+        if(queryParam === undefined) {
+            res.send(documents)
+        } else {
+            const resultsArray = searchQuery(queryParam)
+            if (resultsArray.length !== 0) {
+                res.send(resultsArray);
+            } else {
+                res.status(400).send(`There are no items with searched value`);
+            }
+        }
+    } catch (e) {
+        console.error(e);
+        res.status(500).json({ error: "Internal server error" });
+    }
+})
+
+//GET /documents/:id
+app.get('/documents/:id', (req,res) => {
+    try {
+        const searchId = Number(req.params.id);
+        const resultsArray = documents.find(obj => obj.id === searchId)
+        if (resultsArray !== undefined) {
+            res.send(resultsArray);
+        } else {
+            res.status(400).send(`There are no items with searched id`);
+        }
+    } catch (e) {
+        console.error(e);
+        res.status(500).json({ error: "Internal server error" });
+    }
+})
+
+//POST /search
+app.post('/search', (req,res) => {
+    try {
+       const requestFields = req.body.fields;
+       const queryParam = req.query.q;
+       let resultsArray = [];
+       if(requestFields !== undefined && queryParam !== undefined) {
```

Comment:
Can we just write `if(requestFields && queryParam) {` ?

Reply:
Yeah, we can write like that, it will also work